llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -1778,11 +1778,12 @@ bool SIInsertWaitcnts::generateWaitcntInstBefore(MachineInstr &MI,
           if (IsVGPR) {
             // RAW always needs an s_waitcnt. WAW needs an s_waitcnt unless the
             // previous write and this write are the same type of VMEM
-            // instruction, in which case they're guaranteed to write their
-            // results in order anyway.
+            // instruction, in which case they are (in some architectures)
+            // guaranteed to write their results in order anyway.
```

Comment:
Hi, the change looks good to me, but I have a question:

That means that 2 vector writes of the same type on unrelated memory addresses may introduce a wait, even if they do not alias ? Since we're not considering aliasing information here. 

The same happens for RAW consecutive write/load pairs of instructions. Right ?

Reply:
No this is nothing to do with storing data to memory. We are only talking about loads (or atomic with results) and the order in which they write the loaded data into the result VGPR.